delira/training/base_trainer.py

```diff
@@ -685,14 +705,47 @@ def name(self):
         return os.path.basename(os.path.dirname(os.path.dirname(
             os.path.dirname(self.save_path))))
 
-    def _reinitialize_logging(self, logging_type, logging_kwargs: dict):
-        from delira.logging import TensorboardBackend, VisdomBackend, \
-            BaseBackend, make_logger, register_logger, unregister_logger, \
-            get_available_loggers
+    def _reinitialize_logging(self, logging_type, logging_kwargs: dict,
+                              logging_callback_cls, logging_frequencies,
+                              reduce_types):
+        """
+
+        Parameters
+        ----------
+        logging_type : str or callable
+            the type of logging. If string: it must be one of
+            ["visdom", "tensorboardx"]
+            If callable: it must be a logging handler backend class
+        logging_kwargs : dict
+            dictionary containing all logging keyword arguments
+        logging_callback_cls : class
+            the callback class to create and register for logging
+        logging_callback_cls : class
+            the callback class to create and register for logging
+        logging_frequencies : int or dict
+                specifies how often to log for each key.
+                If int: integer will be applied to all valid keys
+                if dict: should contain a frequency per valid key. Missing keys
+                will be filled with a frequency of 1 (log every time)
+                None is equal to empty dict here.
+        reduce_types : str of FunctionType or dict
```

Comment:
Maybe be a little be more specific: what is reduced inside the logger (also applies to the docstring in the logger)?